root/account/applications/Edit.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * @flow
+ * Copyright (C) 2018 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+import noop from 'lodash/noop';
+
+import FormRow from '../../components/FormRow';
+import FormRowText from '../../components/FormRowText';
+import FormRowURLLong from '../../components/FormRowURLLong';
+import FormSubmit from '../../components/FormSubmit';
+import Layout from '../../layout';
+import {addColon, l} from '../../static/scripts/common/i18n';
+import {Lens, prop, set, compose3} from '../../static/scripts/common/utility/lens';
+import getSelectValue from '../../utility/getSelectValue';
+import hydrate from '../../utility/hydrate';
+
+import type {OauthTypeT, ApplicationFormT} from './types';
+
+type Props = {|
+  +form: ApplicationFormT,
+|};
+
+const EditApplication = ({form}: Props) => (
+  <Layout fullWidth title={l('Edit Application')}>
+    <h1>{l('Edit Application')}</h1>
+    <form method="post">
```

Comment:
Any way you can reuse root/account/applications/RegisterForm.js here? Might need some additional props to change some of the labels, but at least the `onChange` props will be a no-op on the server, as well as the `hydrate` stuff (as long as you don't include client JS, that'll be a no-op too). So it could possibly be reused without too many changes.

I guess it would be good to figure out how to do this since we probably won't want to duplicate all our entity forms between `/create` and `/edit` either, for example.

Reply:
Done in https://github.com/metabrainz/musicbrainz-server/pull/717/commits/cc0bc8af84772df2256e6bdf265e57d77421eac7 by adding `action` and `submitLabel` to the props.